cstar/roms/simulation.py

```diff
@@ -439,57 +514,165 @@ def codebases(self) -> list[ExternalCodeBase]:
+        """Generate and return a ROMSRuntimeSettings object for the simulation.
 
-        ROMS requires a text file containing runtime options to run. This file is
-        typically called `roms.in`, but variations occur. This property finds the
-        appropriate `.in` or `.in_TEMPLATE` file in the `runtime_code` files and
-        returns its path.
+        This property constructs a `ROMSRuntimeSettings` instance from the ROMS `.in`
+        file found in the `runtime_code`, and updates key runtime values based on the
+        current simulation configuration. These include the time step, number of time
+        steps, grid path, initial conditions, forcing datasets, and optionally, MARBL
+        input files.
 
         Returns
         -------
-        Path
-            The path to the ROMS runtime input file.
+        ROMSRuntimeSettings
+            A fully-populated runtime settings object representing the current state
+            of the ROMS simulation configuration.
 
         Raises
         ------
         ValueError
-            - If `runtime_code` is not set.
-            - If multiple `.in` files are found, making the selection ambiguous.
-            - If no `.in` file is found in `runtime_code.files`.
+            If the runtime `.in` file has not been retrieved locally via `setup()` or
+            `runtime_code.get()`.
+
+        Notes
+        -----
+        - This method overrides runtime settings using values from `self.discretization`,
+          `self.model_grid`, `self.initial_conditions`, and all forcing datasets.
+        - If MARBL configuration files are present in `runtime_code.files`, their paths
+          are also included.
         """
 
-        in_files = []
-        if self.runtime_code is None:
+        if self.runtime_code.working_path is None:
             raise ValueError(
-                "ROMSSimulation.runtime_code not set."
-                + " ROMS requires a runtime options file "
-                + "(typically roms.in)"
+                "Cannot access runtime settings without local "
+                + "`.in` file. Call ROMSSimulation.setup() or "
+                + "ROMSSimulation.runtime_code.get() and try again."
             )
 
-        in_files = [
-            fname.replace(".in_TEMPLATE", ".in")
-            for fname in self.runtime_code.files
-            if (fname.endswith(".in") or fname.endswith(".in_TEMPLATE"))
-        ]
-        if len(in_files) > 1:
-            raise ValueError(
-                "Multiple '.in' files found:"
-                + "\n{in_files}"
-                + "\nROMS runtime file choice ambiguous"
+        simulation_runtime_settings = ROMSRuntimeSettings.from_file(
+            self.runtime_code.working_path / self._in_file
+        )
+
+        # Previous modifications
+        # Time step entry
+        simulation_runtime_settings.time_stepping.dt = self.discretization.time_step
+
+        # ntimesteps entry:
```

Comment:
i don't know if it's necessary to have a comment saying "i'm setting the X property" before setting the X property.

Ditto above for `time_stepping.dt`